abd-clam/src/cakes/knn/mod.rs

```diff
@@ -0,0 +1,3 @@
+//! K-Nearest Neighbors (KNN) variants.
+
+pub mod _knn;
```

Comment:
Make this a private module. Then do a public export the enum `KnnAlgorithm`. It should look like:

```rust
mod _knn;

pub use _knn::KnnAlgorithm;
```